components/iosnoop/src/biosnoop.c

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 TiKV Project Authors. Licensed under Apache-2.0.
+
+#include <uapi/linux/ptrace.h>
+#include <linux/blkdev.h>
+
+struct val_t {
+    u64 read;
+    u64 write;
+};
+
+typedef enum {
+    Other,
+    Read,
+    Write,
+    Coprocessor,
+    Flush,
+    Compaction,
+    Replication,
+    LoadBalance,
+    Import,
+    Export,
+} io_type;
+
+BPF_HASH(pidbyreq, struct request *, u32);
+BPF_HASH(typebypid, u32, io_type*);
+BPF_HASH(statsbytype, io_type, struct val_t);
```

Comment:
Since we don't really need to track writes, separate it to R and W to reduce a little contention? Same to IOContext below.

Reply:
I only tracked read bytes before. But seems @yiwu-arbug wants to track both read and write.

Reply:
Yes, I mean we can use ReadIOContext and WriteIOContext to separately retrieve the two.

Reply:
seems unnecessary, we don't need to get them separately

Reply:
Why not? Rate limiter only uses read stats, I thought write stats are only picked up for external metrics.

Reply:
rate limter can get them but not use.